### PR TITLE
Compatible labels with docker-compose.

### PR DESCRIPTION
**Pull request by kunalkushwaha**
Fixes the compatible issue with docker-compose
BugFix #132 

Signed-off-by: Kunal Kushwaha <kushwaha_kunal_v7@lab.ntt.co.jp>

### CODE REVIEW

**Comment by kunalkushwaha** [comment]
This makes service compatible with docker-compose, but may be implementation may not be very correct.

File: docker/container.go [hunk 1]
```diff
@@ -343,6 +358,10 @@ func (c *Container) createContainer(imageName, oldContainer string) (*types.Cont
 	configWrapper.Config.Labels[SERVICE.Str()] = c.service.name
 	configWrapper.Config.Labels[PROJECT.Str()] = c.service.context.Project.Name
 	configWrapper.Config.Labels[HASH.Str()] = c.getHash()
+	// libcompose run command not yet supported, so always "False"
+	configWrapper.Config.Labels[ONEOFF.Str()] = "False"
+	configWrapper.Config.Labels[NUMBER.Str()] = getContainerNumber(c.name)
```
Comment (dnephin) [inline docker/container.go]: I would be better to pass in the number instead of having to parse it from the name.

**Comment by vdemeester** [comment]
LGTM 🐼
I'll add somewhere on my TODO list to add a small integration test that start something with libcompose and stops it using docker-compose.

**Comment by vdemeester** [comment]
@kunalkushwaha Actually, this is not working if I start to scale things up. The following scenario panics for example :

```bash
$ libcompose-cli up -d
# starts two services : simple and another
$ docker-compose ps 
     Name        Command   State   Ports 
----------------------------------------
toto_another_1   top       Up            
toto_simple_1    top       Up 
$ docker-compose scale simple=3
     Name        Command   State   Ports 
----------------------------------------
toto_another_1   top       Up            
toto_simple_1    top       Up            
toto_simple_2    top       Up            
toto_simple_3    top       Up  
$ libcompose-cli ps
WARN[0000] Note: This is an experimental alternate implementation of the Compose CLI (https://github.com/docker/compose) 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x51a282]

goroutine 1 [running]:
github.com/docker/libcompose/docker.(*Container).Info(0xc82035a270, 0xc820310d00, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/libcompose/docker/container.go:80 +0x6b2
github.com/docker/libcompose/docker.(*Service).Info(0xc82032c600, 0xc8202cf800, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/libcompose/docker/service.go:214 +0x159
github.com/docker/libcompose/cli/app.ProjectPs(0xc820210a80, 0xc8201646c0)
        /go/src/github.com/docker/libcompose/cli/app/app.go:54 +0x21a
github.com/docker/libcompose/cli/app.WithProject.func1(0xc8201646c0)
        /go/src/github.com/docker/libcompose/cli/app/app.go:40 +0x154
github.com/docker/libcompose/vendor/github.com/codegangsta/cli.Command.Run(0x95f350, 0x2, 0x0, 0x0, 0x9654a0, 0xf, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/docker/libcompose/vendor/github.com/codegangsta/cli/command.go:113 +0x1082
github.com/docker/libcompose/vendor/github.com/codegangsta/cli.(*App).Run(0xc8202a4270, 0xc82000a120, 0x6, 0x6, 0x0, 0x0)
        /go/src/github.com/docker/libcompose/vendor/github.com/codegangsta/cli/app.go:156 +0xbb1
main.main()
        /go/src/github.com/docker/libcompose/cli/main/main.go:42 +0x15f2

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [syscall]:
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 6 [chan send]:
github.com/docker/libcompose/cli/logger.generateColors()
        /go/src/github.com/docker/libcompose/cli/logger/colors.go:27 +0x182
created by github.com/docker/libcompose/cli/logger.init.1
        /go/src/github.com/docker/libcompose/cli/logger/colors.go:33 +0x2b

goroutine 7 [chan receive]:
github.com/docker/libcompose/project.(*defaultListener).start(0xc820287360)
        /go/src/github.com/docker/libcompose/project/listener.go:47 +0x9e
created by github.com/docker/libcompose/project.NewDefaultListener
        /go/src/github.com/docker/libcompose/project/listener.go:42 +0xb5

goroutine 10 [IO wait]:
net.runtime_pollWait(0x7f15b0a188e0, 0x72, 0xc820012200)
        /usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820300370, 0x72, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820300370, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820300310, 0xc820304000, 0x1000, 0x1000, 0x0, 0x7f15b0a0e050, 0xc820012200)
        /usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820030518, 0xc820304000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/net.go:172 +0xe4
net/http.noteEOFReader.Read(0x7f15b0a189a0, 0xc820030518, 0xc8200147e8, 0xc820304000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc820302340, 0xc820304000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820191c20)
        /usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc820191c20, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc820014790)
        /usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
        /usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 11 [select]:
net/http.(*persistConn).writeLoop(0xc820014790)
        /usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
        /usr/local/go/src/net/http/transport.go:686 +0xc9d
```

**Comment by kunalkushwaha** [comment]
I was able to reproduce this at my end. Occurs if service is created or scaled with `docker-compose` not otherwise.

It is mainly due to `name` label which is used by ``GetContainerByName()`` [function](https://github.com/docker/libcompose/blob/master/docker/functions.go#L35)

Below is grep of NAME Label

``` go
docker/container.go:358:        configWrapper.Config.Labels[NAME.Str()] = c.name
docker/functions.go:35: filterArgs.Add("label", fmt.Sprintf("%s=%s", NAME, name))
docker/functions.go:36: fmt.Println("Label ", NAME , ":", name)
docker/service.go:77:           name := container.Labels[NAME.Str()]
```

Will fix it and update the PR.

**Comment by kunalkushwaha** [comment]
@vdemeester  The issue is fixed and now ``name`` label is removed, and lookup is done on Container's name.

Please review the code as parsing name has some assumption which looks okay in case of compose, but in case I am missing any condition, please let me know :)

File: docker/service.go [hunk 2]
```diff
@@ -74,8 +75,9 @@ func (s *Service) collectContainers() ([]*Container, error) {
 	result := []*Container{}
 
 	for _, container := range containers {
-		name := container.Labels[NAME.Str()]
-		result = append(result, NewContainer(client, name, s))
+		// Compose add "/" before name, so Name[1] will store actaul name.
```
Comment (vdemeester) [inline docker/service.go]: It's not compose but docker that does this (internally).

**Comment by vdemeester** [comment]
LGTM 🐮

**Pull request merged**